packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -376,14 +377,24 @@ describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) =>
 						await queue.add(handle);
 					},
 					async readHandle(): Promise<unknown> {
-						const value = await new Promise((resolve, reject) => {
-							queue
-								.acquire(async (result: FluidObject) => {
-									resolve(result);
-									return ConsensusResult.Release;
-								})
-								.catch((error) => reject(error));
-						});
+						const value = await timeoutAwait(
+							new Promise((resolve, reject) => {
+								const wasNonempty = queue
```

Comment:
if the queue was empty, the previous code does deadlock. I'm not sure how that could happen, but this should make that failure mode more explicit and let us know what we're working with.